drivers/can/can_renesas_ra.c

```diff
@@ -0,0 +1,1114 @@
+	R_CANFD_Type *reg = p_ctrl->p_reg;
+	uint32_t cfdcnctr = reg->CFDC->CTR;
+	int ret = 0;
+
+	if (reg->CFDC->ERFL_b.BOEF == 0) {
+		/* Channel bus-off entry not detected */
+		goto end;
+	}
+
+	reg->CFDC->CTR_b.BOM = 0x00; /* Switch to Normal Bus-Off mode (comply with ISO 11898-1) */
+	reg->CFDC->CTR_b.RTBO = 1;   /* Force channel state to return from bus-off */
+
+	int64_t start_ticks = k_uptime_ticks();
+
+	while (reg->CFDC->ERFL_b.BORF == 0) {
+		if ((k_uptime_ticks() - start_ticks) > timeout.ticks) {
+			ret = -EAGAIN;
+			goto end;
+		}
+	}
+end:
+	reg->CFDC->CTR = cfdcnctr; /* Restore channel configuration */
+	return ret;
+}
+#endif
+
+static inline void can_renesas_ra_call_tx_cb(const struct device *dev, int err)
+{
+	struct can_renesas_ra_data *data = dev->data;
+
+	can_tx_callback_t cb = data->tx_cb;
+
+	if (cb != NULL) {
+		data->tx_cb = NULL;
+		cb(dev, err, data->tx_usr_data);
+		k_sem_give(&data->tx_sem);
+	}
+}
+
+static inline void can_renesas_ra_call_rx_cb(const struct device *dev, can_callback_args_t *p_args)
+{
+	struct can_renesas_ra_data *data = dev->data;
+	struct can_renesas_ra_filter *rx_filter = data->rx_filter;
+	const struct can_renesas_ra_cfg *cfg = dev->config;
+	struct can_frame frame = {
+		.dlc = can_bytes_to_dlc(p_args->frame.data_length_code),
+		.id = p_args->frame.id,
+		.flags = (((p_args->frame.id_mode == CAN_ID_MODE_EXTENDED) ? CAN_FRAME_IDE : 0UL) |
+			  ((p_args->frame.type == CAN_FRAME_TYPE_REMOTE) ? CAN_FRAME_RTR : 0UL) |
+			  ((p_args->frame.options & CANFD_FRAME_OPTION_FD) != 0 ? CAN_FRAME_FDF
+										: 0UL) |
+			  ((p_args->frame.options & CANFD_FRAME_OPTION_ERROR) != 0 ? CAN_FRAME_ESI
+										   : 0UL) |
+			  ((p_args->frame.options & CANFD_FRAME_OPTION_BRS) != 0 ? CAN_FRAME_BRS
+										 : 0UL)),
+	};
+
+	memcpy(frame.data, p_args->frame.data, p_args->frame.data_length_code);
+
+	for (uint32_t i = 0; i < cfg->rx_filter_num; i++) {
+		if (NULL == rx_filter->rx_cb) {
+			rx_filter++;
+			continue; /* this filter has not set yet */
+		}
+
+		if (!can_frame_matches_filter(&frame, &rx_filter->filter)) {
+			rx_filter++;
+			continue; /* filter did not match */
+		}
+
+		rx_filter->rx_cb(dev, &frame, rx_filter->rx_usr_data);
+		break;
+	}
+}
+
+static inline void can_renesas_ra_call_state_change_cb(const struct device *dev,
+						       enum can_state state)
+{
+	struct can_renesas_ra_data *data = dev->data;
+	can_info_t can_info;
+
+	if (FSP_SUCCESS != can_api->infoGet(data->fsp_can.p_ctrl, &can_info)) {
+		LOG_DBG("get state info failed");
+		return;
+	}
+	struct can_bus_err_cnt err_cnt = {
+		.rx_err_cnt = can_info.error_count_receive,
+		.tx_err_cnt = can_info.error_count_transmit,
+	};
+
+	data->common.state_change_cb(dev, state, err_cnt, data->common.state_change_cb_user_data);
+}
+
+static int can_renesas_ra_get_capabilities(const struct device *dev, can_mode_t *cap)
+{
+	ARG_UNUSED(dev);
+	*cap = CAN_MODE_NORMAL | CAN_MODE_LOOPBACK;
+
+#if CONFIG_CAN_FD_MODE
```

Comment:
Please use `#ifdef` all over for conditionals based on `CONFIG_*`:

```suggestion
#ifdef CONFIG_CAN_FD_MODE
```

Reply:
I fixed it